### PR TITLE
Remove the C plugins mechanism

### PR DESCRIPTION
**Pull request by xavierleroy**
The C plugins mechanism was introduced by @lefessan in #668 to support interception for some system calls in the OCaml runtime system.

Two years and a [CVE](https://nvd.nist.gov/vuln/detail/CVE-2017-9772) later, it's unclear that this mechanism is used at all.  Moreover, it complicates the runtime system and is not very general (only a few system functions are instrumented).  There are other ways to intercept system calls that are more general and require no modification to the source code of the runtime system.

Following discussions with other core Caml developers, this PR proposes to just remove this plugin mechanism.  Let the discussion begin!

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -224,9 +223,9 @@ while : ; do
     -with-flambda-invariants|--with-flambda-invariants)
         with_flambda_invariants=true;;
     -with-cplugins|--with-cplugins)
-        with_cplugins=true;;
+        err "--with-cplugins is no longer supported";;  # Transitional?
     -no-cplugins|--no-cplugins)
-        ;; # Ignored for backward compatibility
+        ;; # Ignored for backward compatibility; transitional?
```
Comment (xavierleroy) [inline configure]: Alternatively, the `-with-cplugins` and/or `-no-cplugins` options could be removed altogether, causing a generic error.  I have no opinion.

Comment (dra27) [reply configure]: In general, I think it’s nicer to have the flag preserved with the warning. Apart from anything else, if anything like it is ever reintroduced, it means care is implied not to re-use the old flag name without considering whether the new flag has the same meaning as the old one (largely academic I expect here, of course)

**Comment by dra27** [comment]
Is this running through precheck?

I wholly support this change.

**Comment by xavierleroy** [comment]
This PR went through CI precheck with no errors caused by the changes.  (There was an unrelated error on OpenBSD 32, but it's being fixed separately.)

**Comment by shindere** [comment]
Of course, at the moment this PR is missing an entry in Changes but I
assume you are waiting to make sure everybody is okay before adding it so
this is just a note so that this does not get forgotten.

**Review by shindere — APPROVED** [review]

File: configure (hunk 1 above)
Comment (damiendoligez) [reply configure]: I also think it's better to keep the error (and even remove the "transitional" comments.

**Review by damiendoligez — APPROVED** [review]
Looks good.

**Comment by xavierleroy** [comment]
Thanks for the feedback.  I amended the configure file as suggested and added a Changes entry.  Will merge when CI is green.

**Pull request merged**